client_test.go

```diff
@@ -511,74 +509,73 @@ func Test_Client(t *testing.T) {
 		config.Queues["alternate"] = QueueConfig{MaxWorkers: 10}
 		client := newTestClient(t, bundle.dbPool, config)
 
-		jobStartedChan := make(chan int64)
-
-		type JobArgs struct {
-			JobArgsReflectKind[JobArgs]
-		}
-
-		AddWorker(client.config.Workers, WorkFunc(func(ctx context.Context, job *Job[JobArgs]) error {
-			jobStartedChan <- job.ID
-			return nil
-		}))
-
+		subscribeChan := subscribe(t, client)
 		startClient(ctx, t, client)
 
-		client.producersByQueueName[QueueDefault].testSignals.Init()
-		client.producersByQueueName["alternate"].testSignals.Init()
-
-		insertRes1, err := client.Insert(ctx, &JobArgs{}, nil)
+		insertRes1, err := client.Insert(ctx, &noOpArgs{}, nil)
 		require.NoError(t, err)
 
-		startedJobID := riverinternaltest.WaitOrTimeout(t, jobStartedChan)
-		require.Equal(t, insertRes1.Job.ID, startedJobID)
+		event := riverinternaltest.WaitOrTimeout(t, subscribeChan)
+		require.Equal(t, EventKindJobCompleted, event.Kind)
+		require.Equal(t, insertRes1.Job.ID, event.Job.ID)
 
 		// Pause only the default queue:
 		require.NoError(t, client.QueuePause(ctx, QueueDefault, nil))
-		client.producersByQueueName[QueueDefault].testSignals.Paused.WaitOrTimeout()
+		event = riverinternaltest.WaitOrTimeout(t, subscribeChan)
+		require.Equal(t, EventKindQueuePaused, event.Kind)
+		require.Equal(t, QueueDefault, event.Queue.Name)
```

Comment:
This might depend on whether we get the full queue object in the end, but IMO, feels cleaner to have a single assertion on the full event object. More readable, and produces a better error in case of failure too:

```
		pauseEvent := riverinternaltest.WaitOrTimeout(t, subscribeChan)
		require.Equal(t, &Event{Kind: EventKindQueuePaused, Queue: QueueDefault}, pauseEvent)
```

Reply:
I agree but I think this breaks apart when using the full queue row unfortunately due to the timestamps. Please take a look at the new version when posted though because you may still have ideas to improve.

Reply:
Ended up making this change given the updated approach, thanks!